Documentation/Development.md

```diff
@@ -0,0 +1,124 @@
+# Development
+
+This document contains notes about development and testing of IndexStoreDB.
+
+## Table of Contents
+
+* [Writing Tests](#writing-tests)
+* [Tibs, the "Test Index Build System"](Tibs.md)
+
+## Writing Tests
+
+As much as is practical, all code should be covered by tests. New tests can be added under the `Tests` directory and should use `XCTest`. The rest of this section will describe the additional tools available in the `ISDBTestSupport` module to make it easier to write good and efficient tests.
+
+Most indexer tests follow a pattern:
+
+1. Build and index a test project
+2. Perform index queries
+3. Compare the results against known locations in the test code
+4. (Optional) modify the code and repeat.
+
+### Test Projects (Fixtures)
+
+Index test projects should be put in the `Tests/INPUTS` directory, and use the [Tibs](Tibs.md) build system to define their sources and targets. An example test project might look like:
```

Comment:
Is this `INPUTS` or `Inputs`? There are both in the document.

Reply:
It's `INPUTS`, since I wanted it to be visually obvious it's a special directory.  Fixed.